framework/src/base/MooseVariableBase.C

```diff
@@ -46,9 +46,24 @@ MooseVariableBase::name() const
   return _sys.system().variable(_var_num).name();
 }
 
+std::vector<dof_id_type>
+MooseVariableBase::allDofIndices() const
```

Comment:
To me MooseVariableBase seemed to be a good place for this, I wanted to keep the design similar to dofIndices(). But I understand we would need parallel communication for this. What do you suggest would be a better place?